compiler/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -1,3 +1,27 @@
+//! This file contains the SSA flattening pass - a required pass for ACIR to remove any remaining
+//! control-flow in the singular program-function, resulting in a single block containing the
+//! program logic.
+//!
+//! ACIR/Brillig differences within this pass:
+//!   - This pass is strictly ACIR-only and never mutates brillig functions.
+//!
+//! Conditions:
+//!   - Precondition: There is at most one constrained function (inlining has been performed).
+//!   - Precondition: The 0-1 constrained functions have no loops (unrolling has been performed).
+//!   - Postcondition: The 0-1 constrained functions now consist of only one block where the
```

Comment:
Can we add this to the post-check function in this pass? We can enforce this pretty easily.